src/WrappedGovLst.sol

```diff
@@ -56,44 +81,150 @@ contract WrappedGovLst is ERC20Permit, Ownable {
+  /// @return The minimum amount of fixed liquid staking tokens that would be received.
+  /// @dev Returns a conservative estimate (1 wei less) to account for potential rounding in FIXED_LST transfer.
+  /// The actual amount received may be up to 1 wei more due to FIXED_LST's internal rounding behavior.
+  function previewUnwrapToFixed(uint256 _wrappedAmount) public view virtual returns (uint256) {
+    // At worst 1 wei less than what has been requested will be returned.
+    // The preview will return the minimum amount of assets returned.
+    return _wrappedAmount - 1;
+  }
+
   /// @notice Deposit liquid stake tokens and receive wrapped tokens in exchange.
   /// @param _lstAmountToWrap The quantity of liquid stake tokens to wrap.
   /// @return _wrappedAmount The quantity of wrapped tokens issued to the caller.
   /// @dev The caller must approve at least the amount of tokens to wrap on the lst contract before calling. Amount to
   /// wrap may not be zero.
-  function wrap(uint256 _lstAmountToWrap) external virtual returns (uint256 _wrappedAmount) {
+  function wrapRebasing(uint256 _lstAmountToWrap) external virtual returns (uint256 _wrappedAmount) {
     if (_lstAmountToWrap == 0) {
       revert WrappedGovLst__InvalidAmount();
     }
 
-    uint256 _initialShares = LST.sharesOf(address(this));
     LST.transferFrom(msg.sender, address(this), _lstAmountToWrap);
-    // The amount of tokens issued to the caller is based on number of shares actually issued to the wrapper
-    // token contract. The balance is the number of shares divided by the share scale factor. This means the holder may
-    // lose claim to a slight number of shares the wrapper contract takes possession of, meaning rounding favors the
-    // wrapper, which is desired.
-    _wrappedAmount = (LST.sharesOf(address(this)) - _initialShares) / SHARE_SCALE_FACTOR;
+    FIXED_LST.convertToFixed(_lstAmountToWrap);
+    _wrappedAmount = previewWrapRebasing(_lstAmountToWrap);
     _mint(msg.sender, _wrappedAmount);
 
-    emit Wrapped(msg.sender, _lstAmountToWrap, _wrappedAmount);
+    emit RebasingWrapped(msg.sender, _lstAmountToWrap, _wrappedAmount);
+  }
+
+  /// @notice Deposit underlying stake tokens and receive wrapped tokens in exchange.
+  /// @param _stakeTokensToWrap The quantity of underlying stake tokens to wrap.
+  /// @return _wrappedAmount The quantity of wrapped tokens issued to the caller.
+  /// @dev The caller must approve at least the amount of tokens to wrap on the stake token contract before calling.
+  /// Amount to wrap may not be zero.
+  function wrapUnderlying(uint256 _stakeTokensToWrap) public virtual returns (uint256) {
+    if (_stakeTokensToWrap == 0) {
+      revert WrappedGovLst__InvalidAmount();
+    }
+
+    IERC20 _stakeToken = IERC20(address(LST.STAKE_TOKEN()));
+    _stakeToken.safeTransferFrom(msg.sender, address(this), _stakeTokensToWrap);
+    _stakeToken.approve(address(FIXED_LST), _stakeTokensToWrap);
```

Comment:
Is there any reason we couldn't do a `stakeToken.approve(address(FIXED_LST), type(uint256).max)` in the constructor, to avoid having to make this extra call each time?

Reply:
No, I think that would be fine I will add this change